dev_guide/new_app.adoc

```diff
@@ -39,51 +40,155 @@ application is created.
 
 == Using the CLI
 
-To create an application based on the source code in the current
-directory:
+=== Specifying source code
 
+New-app allows you to create applications using source code from a local or a remote git repository.
+If only a source repository is specified, new-app will try to automatically
+link:#build-strategy-detection[determine the type of build strategy]
+to use (`Docker` or `Source`), and
```

Comment:
link the explanation of the build strategy types:   http://docs.openshift.org/latest/architecture/core_objects/builds.html#build-strategies